Source/AwsCommonRuntimeKit/Utilities.swift

```diff
@@ -145,6 +150,16 @@ extension UnsafePointer: PointerConformance {}
 
 extension UnsafeRawPointer: PointerConformance {}
 
+func withOptionalCString<Result>(
+        to arg1: String?, _ body: (UnsafePointer<Int8>?) -> Result) -> Result {
+    if let arg1 = arg1 {
+        return arg1.withCString { cString in
+            return body(cString)
+        }
+    }
+    return body(nil)
+}
```

Comment:
Thanks, this is a great suggestion. I will update these functions. Out of curiosity, is there a way to write a generic function that takes any number strings and convert them to a byte cursor?

Reply:
There is, but it would require using an array as the input to body like `body: ([aws_byte_cursor]) -> Result` which im not sure is something we want. 

Looking at the usage of these helpers though, we might not need them anyways. 
For example:
```swift
let provider: UnsafeMutablePointer<aws_credentials_provider>? = withByteCursorFromStrings(
                host,
                authToken,
                pathAndQuery
            ) { hostCursor, authTokenCursor, pathAndQueryCursor in
                ecsOptions.host = hostCursor
                ecsOptions.auth_token = authTokenCursor
                ecsOptions.path_and_query = pathAndQueryCursor
                return aws_credentials_provider_new_ecs(allocator.rawValue, &ecsOptions)
            }
```

Could be rewritten as 

```swift
ecsOptions.host = host.withByteCursor { $0 }
ecsOptions.auth_token = authToken.withByteCursor { $0 }
ecsOptions.path_and_query = pathAndQuery.withByteCursor { $0 }

guard let provider = aws_credentials_provider_new_ecs(allocator.rawValue, &ecsOptions) else {
    shutdownCallbackCore.release()
    throw CommonRunTimeError.crtError(CRTError.makeFromLastError())
}
```

Reply:
This is wrong. In Swift, pointers are only valid inside the closure and implementation of bytecursor have pointer. So this will result in an undefined behavior if used this way.

Reply:
Bytecursors should never be stores in a variable and used later. 

Good link: https://developer.apple.com/forums/thread/674633

Reply:
ahhh TIL, thank you!